addons/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/handler/NestBridgeHandler.java

```diff
@@ -237,63 +243,45 @@ private String getExistingOrNewAccessToken() throws InvalidAccessTokenException
         }
     }
 
-    private String jsonFromGetUrl() throws FailedRetrievingNestDataException, InvalidAccessTokenException {
-        try {
-            logger.debug("Fetching data from {}", redirectUrl);
-            return HttpUtil.executeUrl("GET", redirectUrl, getHttpHeaders(), null, null, 5000);
-        } catch (IOException e) {
-            throw new FailedRetrievingNestDataException(e);
-        }
-    }
-
-    private synchronized void startAutomaticRefresh() {
-        if (config == null || config.refreshInterval < MIN_SECONDS_BETWEEN_API_CALLS) {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
-                    "Can not schedule polling job, refresh interval should be set and be more then 60 seconds");
-            return;
-        }
-
-        if (refreshDataJob == null || refreshDataJob.isCancelled()) {
-            refreshDataJob = scheduler.scheduleWithFixedDelay(this::refreshData, 0, config.refreshInterval, SECONDS);
-            logger.debug("Started refreshData job");
-        } else {
-            logger.debug("refreshData job is already scheduled");
-        }
-    }
-
-    private synchronized void stopAutomaticRefresh() {
-        if (refreshDataJob != null && !refreshDataJob.isCancelled()) {
-            refreshDataJob.cancel(true);
-            refreshDataJob = null;
-            logger.debug("Stopping refreshDataJob");
-        } else {
-            logger.debug("refreshDataJob is already stopped");
-        }
-    }
-
     /**
      * @param nestDeviceDataListener The device added listener to add
      */
-    public boolean addDeviceDataListener(NestDeviceDataListener nestDeviceDataListener) {
-        boolean success = listeners.add(nestDeviceDataListener);
-        scheduler.schedule(this::refreshData, 1, SECONDS);
+    public boolean addDeviceDataListener(NestDeviceDataListener listener) {
+        boolean success = listeners.add(listener);
```

Comment:
We seem to ignore the `success` here

Reply:
I think `success` will always be `true` because of the implementation of the `CopyOnWriteArrayList` :smile:  :

```
    /**
     * Appends the specified element to the end of this list.
     *
     * @param e element to be appended to this list
     * @return {@code true} (as specified by {@link Collection#add})
     */
    public boolean add(E e) {
        final ReentrantLock lock = this.lock;
        lock.lock();
        try {
            Object[] elements = getArray();
            int len = elements.length;
            Object[] newElements = Arrays.copyOf(elements, len + 1);
            newElements[len] = e;
            setArray(newElements);
            return true;
        } finally {
            lock.unlock();
        }
    }
```